### PR TITLE
[DML EP] Register pad18

### PR DESCRIPTION
**Pull request by linnealovespie**
### Description

Pad18 adds the `axes` input, which is used to indicate what axes the padding values should be applied to. Add logic to manipulate paddings into DML padding operator inputs.

### CODE REVIEW

File: onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp [hunk 1]
```diff
@@ -41,6 +41,21 @@ namespace OperatorHelper
         }
     }
 
+    void HandleEmptyAxes(
```
Comment (smk2007) [inline onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp:44]: why move up?

Comment (smk2007) [inline onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp:44]: static void

File: onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp [hunk 2]
```diff
@@ -1122,12 +1137,35 @@ namespace OperatorHelper
...
+            ML_CHECK_VALID_ARGUMENT(axes.size() * 2 == padding.size(), "The number of elements in padding should be 2 times the number of axes.");
+            HandleNegativeAxes(axes, dimCount);
+        }
+        else {
+            HandleEmptyAxes(axes, inputShape, false);
+        }
+
+        uint32_t numAxes = gsl::narrow_cast<uint32_t>(axes.size());
+        for(int32_t i = 0; i < axes.size(); i++)
+            {
```
Comment (fdwr) [inline onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp:1163]: Please look it over for funky indentation and spacing, like line 1157, 1162, 1163.

`for(int32_t i = 0; i < axes.size(); i++)`
`for (int32_t i = 0; i < axes.size(); i++)`

`git difftool --dir-diff` is useful for inspecting locally too.

File: onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp [hunk 3]
```diff
@@ -1122,12 +1137,35 @@ namespace OperatorHelper
...
 
-        uint32_t dimCount = gsl::narrow_cast<uint32_t>(padding.size() / 2);
-        m_startPadding.resize(dimCount);
-        m_endPadding.resize(dimCount);
-        std::copy(padding.begin(), padding.begin() + dimCount, m_startPadding.begin());
-        std::copy(padding.begin() + dimCount, padding.begin() + dimCount * 2, m_endPadding.begin());
+        // Handle possible axes input
+        if (opsetVersion >= 18)
+        {
+            if(kernelInformation.IsInputValid(3))
```
Comment (fdwr) [inline onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp:1149]:
```suggestion
            if (kernelInformation.IsInputValid(3))
```
Line 1147 vs 1149

**Review by fdwr — COMMENTED** [review]
Functionally looks valid. Just warrants some cleanup.

**Review by fdwr — DISMISSED** [review]
👍

**Review by fdwr — APPROVED** [review]
Just OperatorKernels.md merge conflicts.

**Pull request merged**